torch/csrc/jit/script/parser.h

```diff
@@ -182,11 +185,58 @@ struct Parser {
       L.expect(end);
     return List<T>::create(r, elements);
   }
+
   Const parseConst() {
     auto range = L.cur().range;
     auto t = L.expect(TK_NUMBER);
     return Const::create(t.range, t.text());
   }
+
+  bool isCharCount(char c, const std::string& str, size_t start, int len) {
+    //count checks from [start, start + len) 
+    return start + len <= str.size() && std::count(str.begin() + start, str.begin() + start + len, c) == len; 
+  }
+
+  std::string parseLexedString(const std::string &str, size_t start) {
+    //invariants: input consists of a valid adjacent python strings and start is the index at beginning 
+    //of a string
+    char quote = str[start];
+    int quote_len = isCharCount(quote, str, start, 3) ? 3 : 1;
+    size_t end = start + quote_len;
+    while(!isCharCount(quote, str, end, quote_len)) {
```

Comment:
If we parsed a single string, then you calculate the body without scanning it. Which will remove the redundant handling of backslashes here.